docs/oauth2.md

```diff
@@ -62,149 +71,79 @@ auth_backends.1 = rabbit_auth_backend_oauth2
-
-The token can be any [JWT token](https://jwt.io/introduction/) which contains the `scope` and `aud` fields. The way the token was issued (such as what grant type was used) is outside of the scope of this plugin.
-
 ### Prerequisites {#prerequisites}
 
 To use this plugin, all RabbitMQ nodes must be
 
 1. [configured to use the rabbit_auth_backend_oauth2 backend](./access-control).
-2. configured with a resource service ID (`resource_server_id`) that matches the scope prefix (e.g. `rabbitmq` in `rabbitmq.read:*/*`).
-3. configured with a signing key used by RabbitMQ to validate the JWT token signatures.
-
-JWT Tokens presented to RabbitMQ for authentication must
-
-1. be digitally signed with either a symmetric or asymmetric key.
-2. have a value in the `aud` field that matches `resource_server_id` value.
-
-
-### Authorization Flow {#authorization-flow-with-scopes}
-
-1. Client requests an `access_token` from the OAuth 2.0 provider,
-2. Token **scope** returned by OAuth 2.0 provider must include RabbitMQ resource scopes that follow a convention used by this plugin: `configure:%2F/foo` means "configure permissions for 'foo' in vhost '/'") (`scope` field can be changed using `extra_scopes_source` in **advanced.config** file.
-3. Client passes the token as password when connecting to a RabbitMQ node. **The username field is ignored**.
-4. The translated permissions are stored as part of the authenticated connection state and used the same way permissions from RabbitMQ's internal database would be used.
-
-## Usage {#usage}
+2. configured with the resource service ID (`resource_server_id`). The RabbitMQ cluster becomes an OAuth 2.0 resource and this is its identifier.
+3. configured with issuer URL of the OAuth 2.0 provider or the JWKS URL or directly with the signing keys the OAuth 2.0 provider will use to sign tokens
 
-The plugin needs a signing key to be configured in order to verify the token's signature. This is the signing key used by the OAuth 2.0 provider to sign the tokens. RabbitMQ supports two types of signing keys: symmetric and asymmetric.
+Here is the minimal configuration to support OAuth 2.0 authentication (*Note*: To enable it in the management plugin you need [additional configuration](./management#minimum-configuration)):
 
-The examples given below uses [Cloud Foundry UAA](https://github.com/cloudfoundry/uaa) as OAuth 2.0 provider.
+```ini
+auth_oauth2.resource_server_id = new_resource_server_id
+auth_oauth2.issuer = https://my-oauth2-provider.com/realm/rabbitmq
+```
 
-To get the signing key from the [OAuth 2.0 provider UAA](https://github.com/cloudfoundry/uaa), use the
-[token_key endpoint](https://docs.cloudfoundry.org/api/uaa/version/4.6.0/index.html#token-key-s)
-or [uaac](https://github.com/cloudfoundry/cf-uaac) (the `uaac signing key` command).
+Based on the above configuration, JWT Tokens presented to RabbitMQ for authentication must:
 
```

Comment:
try not to use words like above and below where possible in docs (for users with accessibility needs), change to "Based on the previous configuration..."